app/controllers/answers_controller.rb

```diff
@@ -26,13 +28,16 @@ def update
     if @answer.update(answer_params)
       head :no_content
     else
-      render json: @answer.errors, status: :unprocessable_entity
+      invalid_request(error_msg)
     end
   end
 
   def destroy
-    @answer.destroy
-    head :no_content
+    if @answer.try(:destroy)
+      head :no_content
```

Comment:
the `head :no_content` returns a response of `status: 204`. So it would not be required here